### PR TITLE
john: Fix for Linuxbrew

### PR DESCRIPTION
**Pull request by carlorosati**
- [x] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/john.rb [hunk 1]
```diff
@@ -14,13 +14,19 @@ class John < Formula
...
     ENV.deparallelize
     arch = MacOS.prefer_64_bit? ? "64" : "sse2"
-    target = "macosx-x86-#{arch}"
-
+    if OS.mac?
+      target = "macosx-x86-#{arch}"
+    elsif OS.linux?
+      target = "generic"
+    end
+
```
Comment (sjackman) [inline Formula/john.rb]:
```
brew style john
C: 29:  1: Trailing whitespace detected.
```
https://travis-ci.org/Linuxbrew/homebrew-core/builds/213226572#L1508

Thanks for your contribution to Linuxbrew, Carlos. Fix this error, and I'll be happy to merge it.

File: Formula/john.rb [hunk 2]
```diff
@@ -14,13 +14,19 @@ class John < Formula
 
   conflicts_with "john-jumbo", :because => "both install the same binaries"
 
-  patch :DATA # Taken from MacPorts, tells john where to find runtime files
+  if OS.mac?
+    patch :DATA # Taken from MacPorts, tells john where to find runtime files
```
Comment (sjackman) [inline Formula/john.rb]: Stylistically, I prefer
```ruby
patch :DATA if OS.mac?
```

File: Formula/john.rb [hunk 3]
```diff
@@ -14,13 +14,19 @@ class John < Formula
...
   def install
     ENV.deparallelize
     arch = MacOS.prefer_64_bit? ? "64" : "sse2"
-    target = "macosx-x86-#{arch}"
-
+    if OS.mac?
+      target = "macosx-x86-#{arch}"
+    elsif OS.linux?
+      target = "generic"
+    end
```
Comment (sjackman) [inline Formula/john.rb]: Stylistically, I prefer
```ruby
target = OS.mac? ? "macosx-x86-#{arch}" : "generic"
```

**Comment by sjackman** [comment]
The preferred commit message is:
`john: Fix for Linuxbrew`

**Comment by sjackman** [comment]
Looks good to me. I'll merge once CI has passed.

**Comment by sjackman** [comment]
Merged. Thanks for your contribution to Linuxbrew, Carlos!

**Comment by carlorosati** [comment]
*Carlo

**Comment by sjackman** [comment]
Thanks for the correction, Carlo. 😄

**Pull request closed without merging**